### PR TITLE
#162949198 Upgrade Django Version

### PR DESCRIPTION
**Pull request by nesh-dev**
#### What does this PR do?
Upgrades Django version from 1.10 to 2.1

#### Description of Task to be completed?
- Update the requirements.txt file with the latest Django version and dependencies.
- Update the url files to match Django 2.1 conventions 
- update the .gitignore file to exclude unneeded migrations
#### How should this be manually tested?
- git clone 
- cd into ah-the-unsullied folder 
- git checkout `ch-upgrade-authors-haven-162949198`
- run on terminal `pip install -r requirements.txt`
- run migrations `python manage.py makemigrations` then `python manage.py migrate`
- create a super user to run the migrations that is `python manage.py createsuperuser`
- run the app `python manage.py runserver`
- test the app on browser or postman. 
#### What are the relevant pivotal tracker stories?
 - [#162949198](https://www.pivotaltracker.com/story/show/162949198)
#### Screenshots (if appropriate)
screenshot 2019-01-07 at 17 58 55

### CODE REVIEW

**Review by indungu — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: authors/settings.py [hunk 1]
```diff
@@ -138,6 +138,10 @@
     'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        'rest_framework.authentication.SessionAuthentication',
+        #'authors.apps.authentication.backends.JWTAuthentication',
+
     ),
+
```
Comment (indungu) [inline authors/settings.py]: Remove unnecessary white spaces.

File: authors/urls.py [hunk 2]
```diff
@@ -13,12 +13,14 @@
     1. Import the include() function: from django.conf.urls import url, include
     2. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
 """
-from django.conf.urls import include, url
+from django.urls import include, path
 from django.contrib import admin
 
+
```
Comment (indungu) [inline authors/urls.py]: Remove unnecessary white spaces.

File: authors/settings.py [hunk 3]
```diff
@@ -138,6 +138,7 @@
     'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        'rest_framework.authentication.SessionAuthentication',
+        #'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (kwanj-k) [inline authors/settings.py]: Good work on the upgrade
Get rid of this commented line though

**Comment by indungu** [comment]
Kindly squash the three commits into a single commit.
Also be sure to edit the typo in the commit message while you are at it.
All in all though, nice work 🎊 .

**Comment by sylviawanjiku** [comment]
The documentation of the PR is good I have managed to set up the application following the steps stated, you could maybe add the steps to create an admin before running the server so that one can at least have a view of the admin site.

**Comment by nesh-dev** [comment]
> The documentation of the PR is good I have managed to set up the application following the steps stated, you could maybe add the steps to create an admin before running the server so that one can at least have a view of the admin site.

I have added this part thanks

File: authors/settings.py [hunk 4]
```diff
@@ -134,10 +134,8 @@
 AUTH_USER_MODEL = 'authentication.User'
 
 REST_FRAMEWORK = {
-    'EXCEPTION_HANDLER': 'authors.apps.core.exceptions.core_exception_handler',
```
Comment (indungu) [inline authors/settings.py]: Is there a particular reason why this exception handler was removed?

File: authors/settings.py [hunk 5]
```diff
@@ -134,10 +134,8 @@
...
-    'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (indungu) [inline authors/settings.py:141]: I notice the default authentication classes were changed, is there a particular reason why this was necessary? Please expound

**Review by indungu — CHANGES_REQUESTED** [review]

File: authors/settings.py (hunk 4 above)
Comment (nesh-dev) [reply authors/settings.py]: This is an issue that has been brought about by changes while rebasing. Let me respond in kind.

File: authors/settings.py (hunk 5 above)
Comment (nesh-dev) [reply authors/settings.py:141]: The authentication class that is used is not set up. I have however commented the line if there is a need to use it while working on authentication.

**Review by kwanj-k — APPROVED** [review]

**Review by sylviawanjiku — APPROVED** [review]

**Review by kelvinwm — APPROVED** [review]

**Review by Allan690 — APPROVED** [review]

File: authors/settings.py (hunk 5 above)
Comment (indungu) [reply authors/settings.py:141]: Okay noted.

**Review by indungu — APPROVED** [review]
LGTM

**Pull request merged**